client/driver/docker.go

```diff
@@ -1353,26 +1375,51 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 		hm := docker.HostMount{
 			Target:   m.Target,
 			Source:   m.Source,
-			Type:     "volume", // Only type supported
+			Type:     m.Type,
 			ReadOnly: m.ReadOnly,
 		}
-		if len(m.VolumeOptions) == 1 {
-			vo := m.VolumeOptions[0]
-			hm.VolumeOptions = &docker.VolumeOptions{
-				NoCopy: vo.NoCopy,
-			}
 
-			if len(vo.DriverConfig) == 1 {
-				dc := vo.DriverConfig[0]
-				hm.VolumeOptions.DriverConfig = docker.VolumeDriverConfig{
-					Name: dc.Name,
+		if hm.Type == "bind" {
+			volumesEnabled := d.config.ReadBoolDefault(dockerVolumesConfigOption, dockerVolumesConfigDefault)
```

Comment:
should there be a different client option for allowing `dockerVolumesConfigOption` and `dockerBindVolumesConfigOption`?

Reply:
I did this to preserve configuration compatibility between the `-v` style implementation and the mount(type=bind) version to ease migration from one to the other.  I think that this could be worth further consideration later though.